### PR TITLE
completed assignment 36

### PR DESCRIPTION
**Pull request by npauzenga**
@Enriikke check it out!

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 Rails.application.routes.draw do
-  resources :posts
+  get 'advertisements/index'
+
+  get 'advertisements/show'
+
+  resources :posts, :advertisements
```
Comment (Enriikke) [inline config/routes.rb]: This line should take care of defining `advertisements/index` and `advertisements/show` for you (in addition to the other RESTful actions :smile:). You can safely remove the two `get` routes defined above.

If you are interested in defining a RESTful resource with just the actions you want, you can do:

```ruby
resources :advertisements, only: [:index, :show]
```

You can also do the opposite filter:

```ruby
resources :advertisements, except: [:new, :create, :edit, :update, :destroy]
```

Keep in mind that you will need to define the `posts` resource separately (in a new line). Using `except` or `only` would affect all of the resources in that declaration, and we don't want that for `posts`. Your routes would look like then look like this:

```ruby
resources :posts
resources :advertisements, only: [:show, :index]
```

**Comment by Enriikke** [comment]
Everything looks awesome @npauzenga! I made a quick comment on your `routes.rb` file. Once you update it let me know and I'll mark you as complete :smile:

**Comment by npauzenga** [comment]
@Enriikke cool, done! Is it generally good practice to specify only the routes you're using, rather than a blanket `resources :advertisements`?

**Comment by Enriikke** [comment]
awesome @npauzenga! :shipit:  

Yes, I would say it's always best to only define the routes you actually need. Say for example, that you left out the route as `resources :advertisements`. This would leave `/advertisements/new` open (among others). Your app would then fail because there is no action or view to handle that endpoint. Requests would receive an **HTTP 500** error (internal server error). On the other hand, if you only expose the endpoints you actually use, then a request to `/advertisements/new` would return an **HTTP 404** error (not found), which is more accurate.

**Pull request merged**